pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/tracker/AnalyticsMediaItemTracker.kt

```diff
@@ -6,150 +6,92 @@ package ch.srgssr.pillarbox.player.tracker
 
 import androidx.media3.common.MediaItem
 import androidx.media3.common.Player
-import androidx.media3.common.Player.PositionInfo
+import androidx.media3.common.Tracks
 import androidx.media3.exoplayer.ExoPlayer
 import androidx.media3.exoplayer.analytics.AnalyticsListener
-import ch.srgssr.pillarbox.player.asset.PillarboxData
-import ch.srgssr.pillarbox.player.extension.getPillarboxDataOrNull
-import ch.srgssr.pillarbox.player.tracker.MediaItemTracker.StopReason
+import ch.srgssr.pillarbox.player.PillarboxExoPlayer
+import ch.srgssr.pillarbox.player.extension.getMediaItemTrackerDataOrNull
 import ch.srgssr.pillarbox.player.utils.DebugLogger
-import ch.srgssr.pillarbox.player.utils.StringUtil
-import kotlin.time.Duration.Companion.milliseconds
 
 /**
- * Custom [CurrentMediaItemPillarboxDataTracker.Callback] to manage analytics.
- *
- * @param player The [Player] whose current [MediaItem] is tracked for analytics.
- * @param mediaItemTrackerProvider The [MediaItemTrackerProvider] that provide new instance of [MediaItemTracker].
+ * Tracks [Player.getCurrentTracks] to handle [MediaItemTrackerData] changes.
+ * @param player The [Player] whose current [Tracks] is tracked for analytics.
  */
 internal class AnalyticsMediaItemTracker(
-    private val player: ExoPlayer,
-    private val mediaItemTrackerProvider: MediaItemTrackerProvider,
-) : CurrentMediaItemPillarboxDataTracker.Callback {
-    private val listener = CurrentMediaItemListener()
+    private val player: PillarboxExoPlayer,
+) : Player.Listener {
 
     /**
      * Trackers are empty if the tracking session is stopped.
      */
-    private var trackers = MediaItemTrackerList()
-    private var currentPillarboxData: PillarboxData? = null
+    private var trackers = mutableListOf<DelegateMediaItemTracker<*>>()
```

Comment:
This can be a `val`, since the list itself is mutable:
```suggestion
    private val trackers = mutableListOf<DelegateMediaItemTracker<*>>()
```

Reply:
Totally